stitches/dev_matchup.py

```diff
@@ -8,155 +8,283 @@
-    col_names = set(input_data.columns)
-    if not (req_cols.issubset(col_names)):
-        raise TypeError(f'Missing columns from "{input_data}".')
-
-    req_exps = {"historical"}
-    exp_set = set(input_data['experiment'].unique())
-
-    if not (req_exps.issubset(exp_set)):
-        raise TypeError(f'{input_data} is missing data for the historical experiment.')
-
-    # Relabel the historical values so that there is a continuous rolling mean between the
-    # historical and future values.
-    # #######################################################
-    # Subset the historical data
-    historical_data = input_data[input_data["experiment"] == "historical"].copy()
-
-    # Create a subset of the future data
-    future_data = input_data[input_data["experiment"] != "historical"].copy()
-    future_scns = set(future_data["experiment"].unique())
-
-    frames = []
-    for scn in future_scns:
-        d = historical_data.copy()
-        d["experiment"] = scn
-        frames.append(d)
-
-    frames.append(future_data)
-    data = pd.concat(frames)
-
-    # Index the data frame by the year, so that the rolling mean respects the years
-    data.index = data['year']
-
-    # Now calculate the rolling mean that uses a centered window.
-    group_by = ['model', 'experiment', 'ensemble', 'variable']
-    rslt = data.groupby(group_by)['value'].rolling(size, center=True).mean().reset_index()
-
-    return rslt
+  # Relabel the historical values so that there is a continuous rolling mean between the
+  # historical and future values.
+  # #######################################################
+  # Subset the historical data
+  historical_data = input_data[input_data["experiment"] == "historical"].copy()
+
+  # Create a subset of the future data
+  future_data = input_data[input_data["experiment"] != "historical"].copy()
+  future_scns = set(future_data["experiment"].unique())
+
+  frames = []
+  for scn in future_scns:
+    d = historical_data.copy()
+    d["experiment"] = scn
+    frames.append(d)
+
+  frames.append(future_data)
+  data = pd.concat(frames)
+
+  # Index the data frame by the year, so that the rolling mean respects the years
+  data.index = data['year']
+
+  # Now calculate the rolling mean that uses a centered window.
+  group_by = ['model', 'experiment', 'ensemble', 'variable']
+  rslt = data.groupby(group_by)['value'].rolling(size, center=True).mean().reset_index()
+
+  return rslt
+
+
+
+def chunk_ts(df, n):
+
+  """ Format a data frame into an array of data frames containing data for n-sized years of successive data.
+
+  :param df:     data frame of climate data to chunk into different periods
+  :type df:     pandas DataFrame
+
+  :param n:    the size of the windows to chunk into separate periods
+  :type n:    int
+
+  :return:    pandas DataFrame identical to df with the addition of a chunk column
+
+  # TODO hmmm do we want this function to also work on ESM data? If so do we need to change this function??
```

Comment:
I think it definitely needs to be able to work on (smoothed) ESM data. I don;t see of the bat any changes we'd have to make though? I think if the `df` input to this function had columns including model name information, they'd still be part of `df` returned, so it should be fine